ephios/core/signup/methods.py

```diff
@@ -78,6 +83,16 @@ def collect_all_qualifications(self) -> set:
     def has_qualifications(self, qualifications):
         return set(qualifications) <= self.collect_all_qualifications()
 
+    def reverse_signup_action(self, shift):
+        raise NotImplementedError
+
+    def reverse_event_detail(self, event):
+        raise NotImplementedError
+
+    @property
+    def icon(self):
+        return mark_safe('<span class="fa fa-user"></span>')
```

Comment:
I'd love to avoid adding more obligatory methods to the AbstractParticipant interface.
But we need some way for Signupviews to know where to redirect users, because I don't think we can easily have the same view, e.g. the EventDetailView implement multiple ways of authenticating (and also there might be different url kwargs, etc...), hence I know doubled those views.

Reply:
What about adding another signal that collects a set of participationtypes including versions of the view? I can't think of a nice architecture right now.